modules/statistics/php/stats_demographic.class.inc

```diff
@@ -163,65 +163,73 @@ class Stats_Demographic extends \NDB_Form
         foreach (\Utility::getProjectList() as $key => $value) {
             $projects[$key] = $value;
         }
-        $currentProject = null;
-        if ($_REQUEST['DemographicProject'] ?? ''
-        ) {
-            $currentProject = htmlspecialchars($_REQUEST['DemographicProject']);
+
+        $demographicProject = $_REQUEST['DemographicProject'] ?? null;
+        $Param_Project      = '';
+
+        if (!empty($demographicProject)) {
+            $demographicProject = htmlspecialchars($demographicProject);
             $this->tpl_data['CurrentProject']
                 = array(
-                   'ID'   => $currentProject,
-                   'Name' => $projects[$currentProject],
+                   'ID'   => $demographicProject,
+                   'Name' => $projects[$demographicProject] ?? '',
                   );
             $Param_Project       ='AND (c.ProjectID IS NULL OR c.ProjectID=:pid) ';
-            $this->params['pid'] = htmlspecialchars($_REQUEST['DemographicProject']);
-        } else {
-            $Param_Project ='';
+            $this->params['pid'] = $demographicProject;
         }
 
         //SUBPROJECTS
-        $subprojList ="";
-        $subprojects = \Utility::getSubprojectsForProject($currentProject);
-        foreach ($subprojects as $key=>$val) {
-            $subprojList .= $key.",";
-        }
-        $subprojList = substr($subprojList, 0, -1);
-        if (!empty($subprojList)) {
-            $suproject_query ="AND s.SubprojectID IN ($subprojList)";
-        } else {
-            $suproject_query ='';
+        $subprojList      = '';
+        $subproject_query = '';
+        $Visits           = \Utility::getExistingVisitLabels($demographicProject);
+        $project          = $demographicProject != '' ? null : $demographicProject;
+        $subprojects      = \Utility::getSubprojectsForProject($project);
+
+        if (!empty($subprojects)) {
+            foreach ($subprojects as $key=>$val) {
+                $subprojList .= "$key,";
+            }
+            $subprojList = substr($subprojList, 0, -1);
+            if (!empty($subprojList)) {
+                $subproject_query = "AND s.SubprojectID IN ($subprojList)";
+            }
         }
 
         //SITES
-        if ($_REQUEST['DemographicSite'] ?? ''
-        ) {
-            $Param_Site          ='AND (c.CenterID IS NULL OR c.CenterID=:sid) ';
-            $this->params['sid'] = htmlspecialchars($_REQUEST['DemographicSite']);
-        } else {
-            $Param_Site = '';
+        $demographicSite = $_REQUEST['DemographicSite'] ?? '';
+        $demographicSite = htmlspecialchars($demographicSite);
```

Comment:
Move this below the `!empty` check

Reply:
Why? if its !empty it will be assigned to `$this->params['sid']`, so we would want it to be escaped before, no?

Reply:
Mostly because it's consistent with the order on line 171 and because there's no use calling the function on an empty string. 

It's not ultimately a huge issue and I won't block the PR just for this.

Reply:
OH i see now, sorry i thought you meant move it below that entire block, not inside the !empty check. I agree.